### ISSUE TITLE
Searching by CallSetIds not supported by HtslibVariantSet

### ISSUE THREAD

**Issue by jeromekelleher** [ga4gh/server#221]
We need to support searching by CallSetIds in the HtslibVariantSet. The implementation should roughly mirror the WormtableVariantSet version: maintain a map of callSetIds to sample indexes, and only retrieve the genotypes for these indexes.

See the pysam source for details and documentation: https://github.com/pysam-developers/pysam/blob/master/pysam/cbcf.pyx

We also require testing for this, and the case in which we do not specify the callSetIds. We should tetst the results using PyVCF. We must test
1. That we get back the correct information for each genotype;
2. We correctly get all genotypes when we specify no callSetIds;
3. Check that we get back the correct calls in the correct order when we specify some callSetIds. See `testSeachByCallSetIds` in `tests/unit/test_backends.py` for similar tests.

**Comment by shajoezhu** [comment]
i am working on this

**Comment by shajoezhu** [comment]
hey @jeromekelleher are callSetIds for variants? or samples?

**Comment by pgrosu** [comment]
Joe, a `CallSet` has its own reference to a `sampleId`:

```
/**
A `CallSet` is a collection of variant calls for a particular sample.
It belongs to a `VariantSet`. This is equivalent to one column in VCF.
*/
record CallSet {

  /** The call set ID. */
  string id;

  /** The call set name. */
  union { null, string } name = null;

  /** The sample this call set's data was generated from. */
  union { null, string } sampleId;
```

Paul

**Comment by shajoezhu** [comment]
cool! thanks. just wanna make sure

**Comment by pgrosu** [comment]
Sure, you're welcome :)  It took me a while as well to clarify the definitions for myself.  Here's a diagram accompanied by a few definitions that clarified things for me (which can be found on http://ga4gh.org/#/api):

![variants](https://cloud.githubusercontent.com/assets/6555937/6646028/7bb83cac-c997-11e4-82eb-e454f790ab3f.png)

A GAVariant represents a change in DNA sequence relative to some reference. For example, a variant could represent a SNP or an insertion. Variants belong to a GAVariantSet. This is equivalent to a row in VCF.

* __VariantSet__ is all the variant information on a specified population
* __Variant__ (row) is a single variant (e.g. "a change A to T at offset 5,227,002 on chr 11", which causes sickle cell anemia)
* __CallSet__ (column) is the genotype information for a single tissue sample
* __Call__ (cell) is the data about a particular sample having a particular variant (e.g. "sample 31415 is 82% likely to have this variant")

Hope it helps,
Paul

**Comment by dcolligan** [comment]
Fixed in #264
